File: herencia/coworking-csharp/Course.OOP.Inheritance.Business/HotDesk.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Course.OOP.Inheritance.Business
{

    /*
    * Esta clase, al ser sealed, termina la jerarquía de clases. Es decir, esta clase no puede ser heredada.
    *
    * Sin embargo, se pueden crear objetos de esta clase sin ningún problema.
    *
    * Al igual que las clases, los métodos también pueden ser sealed.
    *
    * */
    public sealed class HotDesk : Space
    {

        //base() permite invocar el constructor de la clase base.
        public HotDesk(string name): base(name, 1) {}

        public override decimal CalculatePricePerDay()
        {
            //Los puestos flotantes tienen un precio fijo por día.
            return 10;
        }
    }

    //Crear una clase hija de HotDesk generaría error ya que es sealed. Descomenta las siguientes lineas para verlo.
    /*public class AnotherDesk: HotDesk {

        public AnotherDesk(String name): base(name) {}
    }*/
}

File: herencia/coworking-java/src/main/java/course/oop/inheritance/HotDesk.java
package course.oop.inheritance;

import java.math.BigDecimal;

/*
* Esta clase, al ser final, termina la jerarquía de clases. Es decir, esta clase no puede ser heredada.
*
* Sin embargo, se pueden crear objetos de esta clase sin ningún problema.
*
* Al igual que las clases, los métodos también pueden ser final.
*
* */
public final class HotDesk extends Space {

    public HotDesk(String name) {
        //super() permite invocar el constructor de la clase base.
        super(name, 1);
    }

    public BigDecimal calculatePricePerDay() {
        //Los puestos flotantes tienen un precio fijo por día.
        return new BigDecimal(10);
    }
}

//Crear una clase hija de HotDesk generaría error ya que es final. Descomenta las siguientes lineas para verlo.
/*public class AnotherDesk extends HotDesk {

    public AnotherDesk(String name) {
        super(name);
    }
}*/
